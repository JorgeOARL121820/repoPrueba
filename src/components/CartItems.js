import React, { useCallback, useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import Swal from 'sweetalert2';
import { deleteCar, updateCar } from '../actions/actions';

import logo from '../assets/images/logo.png';

export const CartItems = ({ id, precio, origen, destino, duracion,  fecha, hora, horallegada, cantity, nombre, apellidos, direccion, correo, enableButtons = true }) => {

  const [count, setCount] = useState(cantity);

  const dispatch = useDispatch();

  const handleRemove = async () => { 
  

    const result = await Swal.fire({
      title: '¿Desea eliminar el artículo?',
      text: 'El artículo se eliminará de la lista de compras',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Eliminar',
      cancelButtonText: 'Cancelar'
    });
    if (result.value) {

      dispatch(deleteCar(id));

    }

  };

  const handleUpdate = useCallback(async () => {

    if( count === 0 ) {
      const result = await Swal.fire({
        title: '¿Desea eliminar el artículo?',
        text: 'El artículo se eliminará de la lista de compras',
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#3085d6',
        cancelButtonColor: '#d33',
        confirmButtonText: 'Eliminar',
        cancelButtonText: 'Cancelar'
      });

      if (result.value) {

        dispatch(deleteCar(id));
        return;
      }
    }

    dispatch( updateCar( { id, cantity: count } ) );

  }, [count, dispatch, id]);

  useEffect(() => {
    handleUpdate();
  }, [count, handleUpdate]);
  

  const handleConcurrency = (num) => {
    return new Intl.NumberFormat('en-IN', {
        style: 'currency',
        currency: 'USD',
        maximumSignificantDigits: 20}).format(num)
  };

const add = () => {
  if( count >= 10 ) return;

  setCount(count + 1);
};

const remove = () => {
  if( count <= 0 ) return;
  
  setCount(count - 1);
};

  return <>
    <div className="card">
        <div className="col l2 t3 m3">
            <div className="card-image">
                <img src={logo} alt="" width='100%' />
            </div>
        </div>

        <div className="col l12 t12 m12">
            
        <div className="col l12 pull-center">

          {

            enableButtons 
              ? null 
              : (
                <div className="fila col l12">

                  <div className="col l12 t6 m12 pull-left">

                    <h5>
                      Reservante: {nombre} {apellidos}

                      <br />

                      Dirección: {direccion}

                      <br />

                      Correo: {correo}

                      <br />

                      Fecha del vuelo: {fecha}
                    </h5>

                  </div>
            
                </div>
              )

          }
                                <div className="fila">
                                  { enableButtons ? (
                                  <div className="col l12 pull-center">

                                    <h3>
                                      Fecha de vuelo: {fecha}
                                    </h3>

                                  </div>) : null }
                                    <div className="col l2 t4 pull-left">
                                        
                                        <h5>
                                            {'Salida: ' + origen}
                                            <br />
                                            {'Hora: ' + hora}
                                        </h5>

                                    </div>

                                    <div className="col l8 t4">
                                        <i className='bx bxs-plane-alt bx-fade-right' ></i>
                                    </div>

                                    <div className="col l2 t4 pull-right">
                                        
                                        <h5>
                                            {'Llegada: ' + destino}
                                            <br />
                                            {'Hora: ' + horallegada}
                                        </h5>

                                    </div>
                                </div>
                            </div> 

                            <div className="col l12 t12 m12 pull-center">
                                <h5>
                                    { 'Tiempo de viaje: ' + duracion }
                                </h5>
                            </div>

                            <div className="col l12 t12 m12 pull-center">
                                <h5>
                                    { 'Precio: ' + handleConcurrency( precio ) + ' p/p' }
                                </h5>
                            </div>

                            <div className="fila col l12">

                            <div className="fila col l6 t12 m12">

                                <div className='col l6 t12 m12 pull-center'>
                                    <h5> { enableButtons ? '¿Cuántos boletos necesita?' : 'Boletos adquiridos' } </h5>
                                </div>

                                <div className='col l12'></div>

                                <div className="number col l2 t5 m5">
                                  {
                                    enableButtons ?
                                      <button id="restar" onClick={remove}>-</button>
                                    : null
                                  }
                                </div>
                                <div className="col l2 pull-center">
                                    <label id="countValue">{count}</label>
                                </div>
                                <div className="number col l2 t5 m5">
                                    {
                                      enableButtons ?
                                        <button id="agregar" onClick={add}>+</button>
                                      : null
                                    }
                                </div>

                                </div>
                                {
                                  enableButtons ?
                                    <div className="col l6 t12 m12">
                                      <button onClick={handleRemove}>Eliminar del carrito</button>
                                    </div>
                                  : null
                                }

                            </div>


                            <div className="col l12 pull-right">
                              <h3>
                                { ( enableButtons ? 'A pagar por boletos: ' : 'Total pagado: ') + handleConcurrency( precio * count ) }
                              </h3>
                            </div>

        </div>
    </div>

    <hr />
  </>;
};
