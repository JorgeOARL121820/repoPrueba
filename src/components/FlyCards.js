import React, { useState } from 'react';

import logo from '../assets/images/logo.png';
import { CustomButton } from './shop/CustomButton';

export const FlyCards = ({vuelo, origen, destino}) => {

    const [count, setCount] = useState(0);

    const handleConcurrency = (num) => {
        return new Intl.NumberFormat('en-IN', {
            style: 'currency',
            currency: 'USD',
            maximumSignificantDigits: 3}).format(num)
    };

    const add = () => {
        if(count >= 10) return;

        setCount(count + 1);
    };

    const remove = () => {
        if(count === 0) return;

        setCount(count - 1);
    };

    console.log( vuelo );

    return <>
        <div className="card fila">
            <div className="col l2 t2 m3">
                <div className="card-image">
                    <img src={logo} alt="" width="100%"/>
                </div>
            </div>

            <div className="col l12 t12 m12">
                
                <div className="fila">
                    <div className="col l12 t12 m12">

                        <div className="fila">
                            <div className="col l12 pull-center">
                                <div className="fila">
                                    <div className="col l12 pull-center">

                                        <h3>
                                        Fecha de vuelo: {vuelo.fecha}
                                        </h3>

                                    </div>
                                    <div className="col l2 t4 pull-left">
                                        
                                        <h5>
                                            {'Salida: ' + origen}
                                            <br />
                                            {'Hora: ' + vuelo.hora}
                                        </h5>

                                    </div>

                                    <div className="col l8 t4">
                                        <i className='bx bxs-plane-alt bx-fade-right' ></i>
                                    </div>

                                    <div className="col l2 t4 pull-right">
                                        
                                        <h5>
                                            {'Llegada: ' + destino}
                                            <br />
                                            {'Hora: ' + vuelo.horallegada}
                                        </h5>

                                    </div>
                                </div>
                            </div> 

                            <div className="col l6 t12 m12 pull-center">
                                <h5>
                                    { 'Tiempo de viaje: ' + vuelo.duracion }
                                </h5>
                            </div>

                            <div className="col l6 t12 m12 pull-center">
                                <h5>
                                    { 'Precio: ' + handleConcurrency( vuelo.precio ) + ' p/p' }
                                </h5>
                            </div>

                            <div className="col l12">

                            </div>

                        </div>

                    </div>

                    <div className="fila col l6 t12 m12">

                        <div className='col l6 t12 m12 pull-center'>
                            <h5>¿Cuántos boletos necesita?</h5>
                        </div>

                        <div className='col l12'></div>

                        <div className="number col l2 t5 m5">
                            <button onClick={remove}>-</button>
                        </div>
                        <div className="col l2 pull-center">
                            <label>{count}</label>
                        </div>
                        <div className="number col l2 t5 m5">
                            <button onClick={add}>+</button>
                        </div>

                    </div>

                    <div className="col l6 t12 m12">
                        <CustomButton vueloButton={true} text="bx bxs-cart-add" customText='Agregar al carrito' {...vuelo} origenT={origen} destinoT={destino} cantity={count} />
                    </div>
                </div>

            </div>
        </div>

        <hr />
    </>;
};

