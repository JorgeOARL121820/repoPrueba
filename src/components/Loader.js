import React from 'react';

import logo from '../assets/images/loader.gif';

export const Loader = () => {
  return <>
    
    <div className="card">
        
        <div className="pull-center" >
            
            <img className="col l4" src={logo} alt="" />

            <br />

            <h6>Buscando, un momento por favor...</h6>

        </div>

    </div>
  
  </>;
};
