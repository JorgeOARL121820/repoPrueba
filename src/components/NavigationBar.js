import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
// import {NavLink} from 'react-router-dom';
import logo from '../assets/images/logo.png';
import minlogo from '../assets/images/minlogo.png';
import '../styles/navbar.css';

export const NavigationBar = () => {

  const [config, setConfig] = useState({
    isOpen: false,
    logo: minlogo,
    openClass: 'close',
    mode: 'light',
    textMode: 'Modo oscuro'
  });

  const body = document.querySelector('body');
  const principalDiv = document.querySelector('#principalDiv');
  const cards = document.querySelectorAll('.card');
  
  const handleOpenCloseMenu = () => {

    if(config.isOpen) {
      setConfig({
        ...config,
        isOpen: false,
        openClass: '',
        logo: logo
      });
      principalDiv.classList.add('close');
      
    } else {
      setConfig({
        ...config,
        isOpen: true,
        openClass: 'close',
        logo: minlogo
      });
      principalDiv?.classList?.remove('close');
    }

  };

  const handleModeChange = () => {
  
    if(config.mode === 'light') {
      setConfig({
        ...config,
        mode: 'dark',
        textMode: 'Modo claro'
      });
      body.classList.add('dark');
      cards.forEach(card => {
        card.classList.add('dark');
      });
    } else {
      setConfig({
        ...config,
        mode: 'light',
        textMode: 'Modo oscuro'
      });
      body.classList.remove('dark');
      cards.forEach(card => {
        card.classList.remove('dark');
      });
    }


  };


  return (
    <>
    <nav className={'sidebar ' + config.openClass}>
        <header>
            <div className="image-text">
                <span className="image">
                    <img src={config.logo} alt="" />
                </span>
            </div>

            <i className='bx bx-chevron-right toggle' onClick={handleOpenCloseMenu}></i>
        </header>

        <div className="menu-bar">
            <div className="menu">

                <ul className="menu-links">
                    <li className="nav-link">
                        <NavLink to={'/comprar-vuelos'} activeclassname='active'>
                            <i className='bx bxs-plane-alt icon' ></i>
                            <span className="text nav-text">Comprar boletos</span>
                        </NavLink>
                    </li>

                    <li className="nav-link">
                        <NavLink to={'/mis-compras'} activeclassname='active'>
                            <i className='bx bx-history icon' ></i>
                            <span className="text nav-text">Ver mis compras</span>
                        </NavLink>
                    </li>

                </ul>
            </div>

            <div className="bottom-content">
                
                <li className="mode">
                    <div className="sun-moon">
                        <i className='bx bx-moon icon moon'></i>
                        <i className='bx bx-sun icon sun'></i>
                    </div>
                    <span className="mode-text text" > { config.textMode } </span>

                    <div className="toggle-switch" onClick={handleModeChange}>
                        <span className="switch" ></span>
                    </div>
                </li>
                
            </div>
        </div>

    </nav>
    </>
);
};
