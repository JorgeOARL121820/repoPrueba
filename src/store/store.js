import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import { cartReducer } from '../reducers/cartReducer';
import { comprasReducer } from '../reducers/comprasReducers';
import { destinosReducer } from '../reducers/destinosReducers';
import { vuelosReducer } from '../reducers/vuelosReducer';


const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const reducers = combineReducers({
    cart: cartReducer,
    compras: comprasReducer,
    destinos: destinosReducer,
    vuelos: vuelosReducer
});


export const store = createStore(
    reducers,
    composeEnhancers(
        applyMiddleware( thunk )
    )
);