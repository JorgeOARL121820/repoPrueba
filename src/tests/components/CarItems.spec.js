import React from "react";
import { mount } from 'enzyme';


import { CartItems } from "../../components/CartItems";
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import '@testing-library/jest-dom';


describe( 'Prueba de funcionalidad de la seccion de tarjetas del carrito', () => {

    const handleConcurrency = (num) => {
        return new Intl.NumberFormat('en-IN', {
            style: 'currency',
            currency: 'USD',
            maximumSignificantDigits: 20}).format(num)
      };

    const data = { 
        id: 12345678, 
        precio: 25896, 
        origen: 'Morelia', 
        destino: 'Guadalajara', 
        duracion: '5 hrs',  
        fecha: '2022-03-19', 
        hora: "10:00", 
        horallegada: "15:00", 
        enableButtons: true,
        cantity: 5};

    test( 'Comprobar que al dar click en el botón de "+" incremente el número de boletos', () => {
        // const store = store
        const wrapper = mount( 
            <Provider store={store}>
                <CartItems { ...data } />
            </Provider> 
        );

        
        expect( wrapper ).toMatchSnapshot();
        
        wrapper.find( 'button' ).at(1).simulate( 'click' );
        expect( wrapper.find( '#countValue' ).text() ).toBe( (data.cantity + 1).toString() );

    } );

    test( 'Comprobar que al dar click en el bóton de "-" decremente el número de boletos', () => {

        // const store = store
        const wrapper = mount( 
            <Provider store={store}>
                <CartItems { ...data } />
            </Provider> 
        );

        
        expect( wrapper ).toMatchSnapshot();
        
        wrapper.find( 'button' ).at(0).simulate( 'click' );
        expect( wrapper.find( '#countValue' ).text() ).toBe( (data.cantity - 1).toString() );

    } );

    test( 'Comprobar que al dar click en cualquier boton ya sea "+" ó "-" el costo a pagar se modifique correctamente', () => { 

        // const store = store
        const wrapper = mount( 
            <Provider store={store}>
                <CartItems { ...data } />
            </Provider> 
        );

        
        expect( wrapper ).toMatchSnapshot();
        
        //Se reduce en 1 la cantidad de boletos
        wrapper.find( 'button' ).at(0).simulate( 'click' );
        expect( wrapper.find( 'h3' ).at(1).text() ).toBe( `A pagar por boletos: ${ (handleConcurrency(data.precio * (data.cantity - 1))) }` );

        //Como se redujo en 1 la cantidad de boletos, se incrementa en 1, por lo que se debe quedar a la misma cantidad original
        wrapper.find( 'button' ).at(1).simulate( 'click' );
        expect( wrapper.find( 'h3' ).at(1).text() ).toBe( `A pagar por boletos: ${ (handleConcurrency(data.precio * (data.cantity))) }` );


    });

} );