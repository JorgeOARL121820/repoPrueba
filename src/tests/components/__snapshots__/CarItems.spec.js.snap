// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de funcionalidad de la seccion de tarjetas del carrito Comprobar que al dar click en cualquier boton ya sea "+" ó "-" el costo a pagar se modifique correctamente 1`] = `
Array [
  <div
    className="card"
  >
    <div
      className="col l2 t3 m3"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="logo.png"
          width="100%"
        />
      </div>
    </div>
    <div
      className="col l12 t12 m12"
    >
      <div
        className="col l12 pull-center"
      >
        <div
          className="fila"
        >
          <div
            className="col l12 pull-center"
          >
            <h3>
              Fecha de vuelo: 
              2022-03-19
            </h3>
          </div>
          <div
            className="col l2 t4 pull-left"
          >
            <h5>
              Salida: Morelia
              <br />
              Hora: 10:00
            </h5>
          </div>
          <div
            className="col l8 t4"
          >
            <i
              className="bx bxs-plane-alt bx-fade-right"
            />
          </div>
          <div
            className="col l2 t4 pull-right"
          >
            <h5>
              Llegada: Guadalajara
              <br />
              Hora: 15:00
            </h5>
          </div>
        </div>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Tiempo de viaje: 5 hrs
        </h5>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Precio: $25,896 p/p
        </h5>
      </div>
      <div
        className="fila col l12"
      >
        <div
          className="fila col l6 t12 m12"
        >
          <div
            className="col l6 t12 m12 pull-center"
          >
            <h5>
               
              ¿Cuántos boletos necesita?
               
            </h5>
          </div>
          <div
            className="col l12"
          />
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="restar"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="col l2 pull-center"
          >
            <label
              id="countValue"
            >
              5
            </label>
          </div>
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="agregar"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
        <div
          className="col l6 t12 m12"
        >
          <button
            onClick={[Function]}
          >
            Eliminar del carrito
          </button>
        </div>
      </div>
      <div
        className="col l12 pull-right"
      >
        <h3>
          A pagar por boletos: $1,29,480
        </h3>
      </div>
    </div>
  </div>,
  <hr />,
]
`;

exports[`Prueba de funcionalidad de la seccion de tarjetas del carrito Comprobar que al dar click en el botón de "+" incremente el número de boletos 1`] = `
Array [
  <div
    className="card"
  >
    <div
      className="col l2 t3 m3"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="logo.png"
          width="100%"
        />
      </div>
    </div>
    <div
      className="col l12 t12 m12"
    >
      <div
        className="col l12 pull-center"
      >
        <div
          className="fila"
        >
          <div
            className="col l12 pull-center"
          >
            <h3>
              Fecha de vuelo: 
              2022-03-19
            </h3>
          </div>
          <div
            className="col l2 t4 pull-left"
          >
            <h5>
              Salida: Morelia
              <br />
              Hora: 10:00
            </h5>
          </div>
          <div
            className="col l8 t4"
          >
            <i
              className="bx bxs-plane-alt bx-fade-right"
            />
          </div>
          <div
            className="col l2 t4 pull-right"
          >
            <h5>
              Llegada: Guadalajara
              <br />
              Hora: 15:00
            </h5>
          </div>
        </div>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Tiempo de viaje: 5 hrs
        </h5>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Precio: $25,896 p/p
        </h5>
      </div>
      <div
        className="fila col l12"
      >
        <div
          className="fila col l6 t12 m12"
        >
          <div
            className="col l6 t12 m12 pull-center"
          >
            <h5>
               
              ¿Cuántos boletos necesita?
               
            </h5>
          </div>
          <div
            className="col l12"
          />
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="restar"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="col l2 pull-center"
          >
            <label
              id="countValue"
            >
              5
            </label>
          </div>
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="agregar"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
        <div
          className="col l6 t12 m12"
        >
          <button
            onClick={[Function]}
          >
            Eliminar del carrito
          </button>
        </div>
      </div>
      <div
        className="col l12 pull-right"
      >
        <h3>
          A pagar por boletos: $1,29,480
        </h3>
      </div>
    </div>
  </div>,
  <hr />,
]
`;

exports[`Prueba de funcionalidad de la seccion de tarjetas del carrito Comprobar que al dar click en el bóton de "-" decremente el número de boletos 1`] = `
Array [
  <div
    className="card"
  >
    <div
      className="col l2 t3 m3"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="logo.png"
          width="100%"
        />
      </div>
    </div>
    <div
      className="col l12 t12 m12"
    >
      <div
        className="col l12 pull-center"
      >
        <div
          className="fila"
        >
          <div
            className="col l12 pull-center"
          >
            <h3>
              Fecha de vuelo: 
              2022-03-19
            </h3>
          </div>
          <div
            className="col l2 t4 pull-left"
          >
            <h5>
              Salida: Morelia
              <br />
              Hora: 10:00
            </h5>
          </div>
          <div
            className="col l8 t4"
          >
            <i
              className="bx bxs-plane-alt bx-fade-right"
            />
          </div>
          <div
            className="col l2 t4 pull-right"
          >
            <h5>
              Llegada: Guadalajara
              <br />
              Hora: 15:00
            </h5>
          </div>
        </div>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Tiempo de viaje: 5 hrs
        </h5>
      </div>
      <div
        className="col l12 t12 m12 pull-center"
      >
        <h5>
          Precio: $25,896 p/p
        </h5>
      </div>
      <div
        className="fila col l12"
      >
        <div
          className="fila col l6 t12 m12"
        >
          <div
            className="col l6 t12 m12 pull-center"
          >
            <h5>
               
              ¿Cuántos boletos necesita?
               
            </h5>
          </div>
          <div
            className="col l12"
          />
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="restar"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <div
            className="col l2 pull-center"
          >
            <label
              id="countValue"
            >
              5
            </label>
          </div>
          <div
            className="number col l2 t5 m5"
          >
            <button
              id="agregar"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
        <div
          className="col l6 t12 m12"
        >
          <button
            onClick={[Function]}
          >
            Eliminar del carrito
          </button>
        </div>
      </div>
      <div
        className="col l12 pull-right"
      >
        <h3>
          A pagar por boletos: $1,29,480
        </h3>
      </div>
    </div>
  </div>,
  <hr />,
]
`;
